pkg/services/vmoperator/vmopmachine.go

```diff
@@ -270,6 +283,76 @@ func (v *VmopMachineService) ReconcileNormal(ctx context.Context, machineCtx cap
 	return false, nil
 }
 
+const (
+	maxNameLength = 63
+)
+
+// Note: Inlining these functions from sprig to avoid introducing a dependency.
+var nameTemplateFuncs = map[string]any{
+	"trimSuffix": func(a, b string) string { return strings.TrimSuffix(b, a) },
+	"trunc": func(c int, s string) string {
+		if c < 0 && len(s)+c > 0 {
+			return s[len(s)+c:]
+		}
+		if c >= 0 && len(s) > c {
+			return s[:c]
+		}
+		return s
+	},
+}
+
+var nameTpl = template.New("name generator").Funcs(nameTemplateFuncs).Option("missingkey=error")
+
+// virtualMachineObjectKey returns the object key of the VirtualMachine.
+// Part of this is generating the name of the VirtualMachine based on the naming strategy.
+func virtualMachineObjectKey(machineName, machineNamespace string, namingStrategy *vmwarev1.VirtualMachineNamingStrategy) (*client.ObjectKey, error) {
+	name, err := GenerateVirtualMachineName(machineName, namingStrategy)
+	if err != nil {
+		return nil, err
+	}
+
+	return &client.ObjectKey{
+		Namespace: machineNamespace,
+		Name:      name,
+	}, nil
+}
+
+// GenerateVirtualMachineName generates the name of a VirtualMachine based on the naming strategy.
+func GenerateVirtualMachineName(machineName string, namingStrategy *vmwarev1.VirtualMachineNamingStrategy) (string, error) {
+	// Per default the name of the VirtualMachine should be equal to the Machine name (this is the same as "{{ .machine.name }}")
+	if namingStrategy == nil || namingStrategy.Template == nil {
+		// Note: No need to trim to max length in this case as valid Machine names will also be valid VirtualMachine names.
+		return machineName, nil
+	}
+
+	nameTemplate := *namingStrategy.Template
+	data := map[string]interface{}{
+		"machine": map[string]interface{}{
+			"name": machineName,
+		},
+	}
+
+	tpl, err := nameTpl.Parse(nameTemplate)
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to generate name for VirtualMachine: failed to parse namingStrategy.template %q", nameTemplate)
+	}
+
+	var buf bytes.Buffer
+	if err := tpl.Execute(&buf, data); err != nil {
+		return "", errors.Wrap(err, "failed to generate name for VirtualMachine")
+	}
+
+	name := buf.String()
+
+	// If the name exceeds the maxNameLength, trim to maxNameLength.
+	// Note: we're not adding a random suffix as the name has to be deterministic.
+	if len(name) > maxNameLength {
+		name = name[:maxNameLength]
```

Comment:
Thinking about this.
Truncating without introducing a sort of randomness might lead to conflicts...
wdyt?

Reply:
If I introduce randomness here we have to do a lot more things. I can't start using a different random name for create / get. 

We can consider this but it requires some logic that we are able to get VirtualMachines that we created before so that we are using the Generate func 100% only on create (which is currently not the case)

Reply:
Added a comment to the API field